src/modules/frequency/application/dtos/class-attendance.dto.ts

```diff
@@ -0,0 +1,150 @@
+import { ApiProperty } from "@nestjs/swagger";
+import { Transform, Type } from "class-transformer";
+import {
+  IsArray,
+  IsDate,
+  IsEnum,
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  ValidateNested,
+} from "class-validator";
+import { FrequencyStatus } from "src/common/enums/domain.enums";
+export class StudentClassAttendanceItemDTO {
+  @ApiProperty({
+    example: 11,
+    description: "The unique ID of the frequency record"
+  })
+  frequencyId: number;
+
+  @ApiProperty({
+    example: 1,
+    description: "The ID of the student"
+  })
+  studentId: number;
+
+  @ApiProperty({
+    example: "João Silva",
+    description: "The student's full name"
+  })
+  studentFullName: string;
+
+  @ApiProperty({
+    example: "90.00%",
+    description: "The student's attendance percentage for this class"
+  })
+  attendancePercentage: string;
+
+  @ApiProperty({
+    example: "AUSENTE",
+    description: "The attendance status for this specific date",
+    enum: ["PRESENTE", "AUSENTE"]
+  })
+  status: string;
+
+  @ApiProperty({
+    example: "ATESTADO-MEDICO",
+    description: "Additional notes about the student's attendance",
+    nullable: true
+  })
+  notes: string | null;
+}
+
+export class StudentListByClassAndDateResponseDTO {
+  @ApiProperty({
+    example: 2,
+    description: "The ID of the class"
+  })
+  classId: number;
+  
+  @ApiProperty({
+    example: "2025-09-02",
+    description: "The date of the attendance record"
+  })
+  date: String;
+  
+  @ApiProperty({
+    type: [StudentClassAttendanceItemDTO],
+    description: "List of students with their attendance details"
+  })
+  studentList: StudentClassAttendanceItemDTO[];
+}
+
+export class PostClassAttendanceDTO {
+  @IsNumber()
+  @Type(() => Number)
+  classId: number;
+  @IsDate()
+  @Transform(({ value }) => new Date(value))
+  @ApiProperty({
+    example: "2025-09-11",
+    description: "The attendance class date",
+  })
+  date: Date;
+}
+export interface EnrolledStudentDTO {
+  id: number;
+}
+
+export class UpdateAttendanceItemDTO {
+  @ApiProperty({
+    example: 19,
+    description: "The ID of the frequency record to update"
+  })
+  @IsNumber()
+  @IsNotEmpty()
+  frequencyId: number;
+
+  @ApiProperty({
+    example: 2,
+    description: "The ID of the student"
+  })
+  @IsNumber()
+  @IsNotEmpty()
+  studentId: number;
+
+  @ApiProperty({
+    example: "PRESENTE",
+    description: "The updated attendance status",
+    enum: ["PRESENTE", "AUSENTE"]
+  })
+  @IsEnum(FrequencyStatus)
+  @IsNotEmpty()
+  status: FrequencyStatus;
+
+  @ApiProperty({
+    example: "ATESTADO-MEDICO",
+    description: "Additional notes about the attendance",
+    nullable: true
+  })
+  @IsOptional()
+  notes: string | null;
```

Comment:
See previous notes comment
```suggestion
  notes: "A" | "B";
```

Reply:
A princípio, `null` deverá ser aceito também, dado que `notes` só existirão, se houver ausência.

Reply:
Sim, tem razão, acabei removendo o `null` sem querer. O objetivo era tornar essa string em uma constante/enum